js/task-controller.js

```diff
@@ -0,0 +1,105 @@
+export default class TaskController {
+
+    _tasks = [];
+
+    currentTasks_container = document.querySelector("#currentTasks");
+    completedTasks_container = document.querySelector("#completedTasks");
+
+    setTasks(newTasks) {
+        this._tasks = newTasks;
+    }
+
+    getTasks() {
+        return this._tasks;
+    }
+
+    addTask(task) {
+        const newTasks = this.getTasks();
+        newTasks.push(task);
+        this.setTasks(newTasks);
+    }
+
+    updateTask(props) {
+        const currentTasks = this.getTasks();
+
+
+        const { id, title, text, priority } = props;
+
+        const newTasks = currentTasks.map(task => {
+            if (task.id == id) {
+                task.title = title;
+                task.text = text;
+                task.priority = priority;
+            }
+            return task;
+        });
+
+        this.setTasks(newTasks);
+    }
+
+    completeTask(id) {
+        const currentTasks = this.getTasks();
+
+        const newTasks = currentTasks.map(task => {
+            if (task.id == id) {
+                task.isCompleted = !task.isCompleted;
+            }
+            return task;
+        });
+
+        this.setTasks(newTasks);
+    }
+
+    deleteTask(id) {
+
+        const currentTasks = this.getTasks();
+        const newTasks = currentTasks.filter(task => task.id != id);
+        this.setTasks(newTasks);
+    }
+
+    renderTasks(sort = "asc") {
+
+        const tasks = this.getTasks();
+        const sortedTasks = this.sortTasks(tasks, sort);
+
+        this.currentTasks_container.innerHTML = "";
+        this.completedTasks_container.innerHTML = "";
+
+        sortedTasks.forEach(task => {
+            if (!task.isCompleted) {
+                this.currentTasks_container.append(task.render())
+            } else {
+                this.completedTasks_container.append(task.render());
+            }
+        });
+
+        this.setHeaders();
+    }
+
+    sortTasks(arr, sort) {
+
+        const sortArr = arr.sort(sortByDate);
+        return (sort === "asc") ? sortArr : sortArr.reverse();
+
+        function sortByDate(a, b) {
+            return new Date(b.date) - new Date(a.date);
+        };
```

Comment:
Please extract method outside and if it's used only for your purpose mark it as private via `#sortByDate` or `_sortByDate`.
It's your choice.

Reply:
made it private like `_sortByDate`
```suggestion
     sortTasks(arr, sort) {
        const sortArr = arr.sort(this._sortByDate);
        return (sort === "asc") ? sortArr : sortArr.reverse();
      };
  
      _sortByDate(a, b) {
          return new Date(b.date) - new Date(a.date);
      };
```